test/Microsoft.DotNet.Configurer.UnitTests/GivenAFirstTimeUseNoticeSentinel.cs

```diff
@@ -0,0 +1,175 @@
+        {
+            var fileSystemMock = _fileSystemMockBuilder.Build();
+            var firstTimeUseNoticeSentinel =
+                new FirstTimeUseNoticeSentinel(
+                    DOTNET_USER_PROFILE_FOLDER_PATH,
+                    fileSystemMock.File,
+                    fileSystemMock.Directory);
+
+            firstTimeUseNoticeSentinel.Exists().Should().BeFalse();
+
+            firstTimeUseNoticeSentinel.CreateIfNotExists();
+
+            firstTimeUseNoticeSentinel.Exists().Should().BeTrue();
+        }
+
+        [Fact]
+        public void ItDoesNotCreateTheSentinelAgainIfItAlreadyExistsInTheDotnetUserProfileFolderPath()
+        {
+            const string contentToValidateSentinalWasNotReplaced = "some string";
+            var sentinel = Path.Combine(DOTNET_USER_PROFILE_FOLDER_PATH, FirstTimeUseNoticeSentinel.SENTINEL);
+            _fileSystemMockBuilder.AddFile(sentinel, contentToValidateSentinalWasNotReplaced);
+
+            var fileSystemMock = _fileSystemMockBuilder.Build();
+
+            var firstTimeUseNoticeSentinel =
+                new FirstTimeUseNoticeSentinel(
+                    DOTNET_USER_PROFILE_FOLDER_PATH,
+                    fileSystemMock.File,
+                    fileSystemMock.Directory);
+
+            firstTimeUseNoticeSentinel.Exists().Should().BeTrue();
+
+            firstTimeUseNoticeSentinel.CreateIfNotExists();
+
+            fileSystemMock.File.ReadAllText(sentinel).Should().Be(contentToValidateSentinalWasNotReplaced);
+        }
+
+        [Fact]
+        public void ItCreatesTheDotnetUserProfileFolderIfItDoesNotExistAlreadyWhenCreatingTheSentinel()
+        {
+            var fileSystemMock = _fileSystemMockBuilder.Build();
+            var directoryMock = new DirectoryMock();
+            var firstTimeUseNoticeSentinel =
+                new FirstTimeUseNoticeSentinel(
+                    DOTNET_USER_PROFILE_FOLDER_PATH,
+                    fileSystemMock.File,
+                    directoryMock);
+
+            firstTimeUseNoticeSentinel.CreateIfNotExists();
+
+            directoryMock.Exists(DOTNET_USER_PROFILE_FOLDER_PATH).Should().BeTrue();
+            directoryMock.CreateDirectoryInvoked.Should().BeTrue();
+        }
+
+        [Fact]
+        public void ItDoesNotAttemptToCreateTheDotnetUserProfileFolderIfItAlreadyExistsWhenCreatingTheSentinel()
+        {
+            var fileSystemMock = _fileSystemMockBuilder.Build();
+            var directoryMock = new DirectoryMock(new List<string> { DOTNET_USER_PROFILE_FOLDER_PATH });
+            var firstTimeUseNoticeSentinel =
+                new FirstTimeUseNoticeSentinel(
+                    DOTNET_USER_PROFILE_FOLDER_PATH,
+                    fileSystemMock.File,
+                    directoryMock);
+
+            firstTimeUseNoticeSentinel.CreateIfNotExists();
+
+            directoryMock.CreateDirectoryInvoked.Should().BeFalse();
+        }
+
+        private class DirectoryMock : IDirectory
```

Comment:
Nit: it should be called fake https://www.martinfowler.com/articles/mocksArentStubs.html

Dummy objects are passed around but never actually used. Usually they are just used to fill parameter lists.
**Fake** objects actually have working implementations, but usually take some shortcut which makes them not suitable for production (an in memory database is a good example).
**Stubs** provide canned answers to calls made during the test, usually not responding at all to anything outside what's programmed in for the test.
**Spies** are stubs that also record some information based on how they were called. One form of this might be an email service that records how many messages it was sent.
**Mocks** are what we are talking about here: objects pre-programmed with expectations which form a specification of the calls they are expected to receive.